main.go

```diff
@@ -158,6 +159,8 @@ func main() {
 	daemonAPI := daemonset.NewCreator(client, kernelLabel, scheme)
 	kernelAPI := module.NewKernelMapper()
 	moduleStatusUpdaterAPI := statusupdater.NewModuleStatusUpdater(client, daemonAPI, metricsAPI)
+	preflightStatusUpdaterAPI := statusupdater.NewPreflightStatusUpdater(client)
+	preflightAPI := preflight.NewPreflightAPI(client, registryAPI, kernelAPI)
```

Comment:
Why do we have 2 different interfaces for Preflight?

Reply:
we have interface for updating the statuses of the KMM objects (module, preflight, maybe later for hub/spoke), and we have interface for preflight processing

Reply:
Why don't we use a single interface? or a single interface that embed both smaller interfaces if you prefer the separation?
I just wonder why the Reconciler need to different interfaces for manipulating Preflight.

Reply:
that's was a discussion that we had when creating/updating statusupdater package. we use this package as an updater for all the object of KMM, and it provides different interfaces for different objects